app/sidekiq/file_extraction_worker.rb

```diff
@@ -8,21 +8,41 @@ class FileExtractionWorker
 
   def perform(extraction_job_id)
     initialize_instance_variables(extraction_job_id)
+    reset_harvest_report_if_needed
```

Comment:
you can probably drop the if_needed so it just calls it and it happens if it's required or not